autoload/go/debug.vim

```diff
@@ -391,20 +391,33 @@ function! go#debug#StartWith(...) abort
   endif
 
   try
+    " Run dlv from a temporary directory so it won't put the binary in the
+    " current dir. We pass --wd= so the binary is still run from the current
+    " dir.
+    let original_dir = getcwd()
+    let tmp = fnamemodify(tempname(), ':h')
```

Comment:
```
let tempfile = tempname()
let tmpdir = fnamemodify(tempfile, ':h')
call delete(tempfile)
```

Reply:
Alright, I'll change that in a minute :+1:

Would you mind rebasing your `godebug` branch on the latest master btw? Then I have the bugfix for `go#package#FromPath()` so I can test it to make sure I didn't make a stupid mistake :-)

Reply:
Looking at this again, I think your intent was to make sure that Vim creates a *directory* in `/tmp/<random>`? It already does that; the return value from `tempname()` is `/tmp/<random>/<n>`, where `<random>` seems stable for every Vim session and `n` is a sequential number.

So I think that the current version already does what you want? Write file to `/tmp/<random>/<pkgname>`?

Reply:
tempname() doesn't create directory (at least on Windows)

Reply:
Sorry tempname() doesn't create file too.

Reply:
How about this?

```vim
let tmpdir = filter([$TEMP, $TMP, $TMPDIR], '!empty(v:val)')[0] . '/' . sha256(bufname(''))
call mkdir(tmpdir, "p", 0700)
```

Reply:
Ah, I didn't know it behaves different on Windows. I updated it with something that should work on both platforms (but I didn't test on Windows, as I don't have a Windows computer!)